packages/autotest/src/autotest/AutoTest.ts

```diff
@@ -305,116 +304,45 @@ export abstract class AutoTest implements IAutoTest {
-                    Log.trace("AutoTest::invokeContainer(..) - output: " + JSON.stringify(output, null, 2));
-
-                    record = {
-                        delivId,
-                        repoId,
-                        commitURL,
-                        commitSHA,
-                        input,
-                        output
-                    };
-
-                    // POST the grade to Class Portal
-                    // NOTE: it is ok for for this to be here, but the backend should consider
-                    // whether or not to honour it (e.g., based on course config or deadlines)
-
-                    let score = -1;
-                    if (output.report !== null && typeof output.report.scoreOverall !== "undefined") {
-                        score = output.report.scoreOverall;
-                    }
-                    const gradePayload: AutoTestGradeTransport = {
-                        delivId,
-                        repoId,
-                        repoURL,
-                        score,
-
-                        urlName: repoId, // could be a short SHA, but this seems better
-                        URL: commitURL,
-
-                        comment: '', // output.report.feedback,   // this only makes sense if we can render markdown
-                        timestamp,
-                        custom: {}
-                    };
-
-                    if (output.postbackOnComplete === false) {
-                        await this.classPortal.sendGrade(gradePayload); // this is just the Grade record, not the Report record
-                    } else {
-                        // grade not sent; if postback is true we must have compile / lint problem
-                    }
-                } catch (err) {
-                    Log.warn("AutoTest::invokeContainer(..) - ERROR for SHA: " + input.target.commitSHA +
-                        "; ERROR sending grade: " + err);
-                }
-            } else {
-                Log.info("AutoTest::invokeContainer(..) - TEST CONFIG: Running MockGrader");
-                const grader = new MockGrader(input);
-                record = await grader.execute();
+            const githubHost = Config.getInstance().getProp(ConfigKey.githubHost);
+            const org = Config.getInstance().getProp(ConfigKey.org);
+            const repoId = input.target.repoId;
+            const gradePayload: AutoTestGradeTransport = {
+                delivId: input.delivId,
+                repoId,
+                repoURL: `${githubHost}/${org}/${input.target.repoId}`,
+                score,
+                urlName: repoId,
+                URL: input.target.commitURL,
+                comment: '',
+                timestamp: input.target.timestamp,
+                custom: {}
+            };
+
+            if (record.output.postbackOnComplete === false) {
+                await this.classPortal.sendGrade(gradePayload); // this is just the Grade record, not the Report record
```

Comment:
I don't like this guard; we should probably just send the grade regardless and let the backend figure out if it wants to accept it or not.